src/components/EpisodeApp.tsx

```diff
@@ -0,0 +1,83 @@
+import { useState, useEffect } from "react";
+import {
+  searchName,
+  searchSummary,
+  searchEpisodeId,
+} from "../core/contentFilter";
+import { getEpisodes, IEpisode } from "../core/fetchEpisodes";
+import { IShow } from "../core/fetchShows";
+import SearchBox from "./SearchBox";
+import FilterDropDown from "./FilterDropDown";
+import EpisodeList from "./EpisodeList";
+
+interface EpisodeAppProps {
+  showFilter: string;
+  setShowFilter(showCode: string): void;
+  shows: IShow[];
+}
+
+export default function EpisodeApp({
+  showFilter,
+  setShowFilter,
+  shows,
+}: EpisodeAppProps) {
+  const [episodes, setEpisodes] = useState<IEpisode[]>([]);
+  const [searchInput, setSearchInput] = useState("");
+  const [episodeFilter, setEpisodeFilter] = useState("");
+
+  useEffect(() => {
+    async function fetchEpisodeData() {
+      const episodeData = await getEpisodes(showFilter);
+      setEpisodes(episodeData);
+      setEpisodeFilter("");
+      setSearchInput("");
+    }
+
+    fetchEpisodeData();
+  }, [showFilter]);
+
+  const filteredEpisodes = [
+    ...new Set<IEpisode>([
+      ...episodes.filter(searchName(searchInput)),
+      ...episodes.filter(searchSummary(searchInput)),
+    ]),
+  ].filter(searchEpisodeId(episodeFilter));
```

Comment:
Difficult to understand it, we spent over 10 minutes going by all the code involved in this snippet

Very smart code, but readability could be improved: the functions' name can be clearer by signaling they return a function

Another thing to consider is separating the filters from lines 41 and 42 to variables

In case there is episodeFilter applied, it would be more efficient if you were going through only one episode, not all of the episodes